src/utils_request.js

```diff
@@ -0,0 +1,272 @@
+ *  class and it should return `true` if request should be aborted, or `false` otherwise.
+ * @param [options.throwOnHttpError=false]
+ *  If set to true function throws and error on 4XX and 5XX response codes.
+ * @return {{ response, body }}
+ *  Returns an object with two properties: `response` is the instance of
+ *  Node's [`http.IncomingMessage`](https://nodejs.org/api/http.html#http_class_http_incomingmessage) class,
+ *  `body` is a `String`, `Buffer` or `Object`, depending on the `encoding` and `json` options.
+ */
+export const requestBetter = async (options) => {
+    return new Promise((resolve, reject) => {
+        // Using the streaming API of Request to be able to
+        // handle the response based on headers receieved.
+        const defaultOptions = {
+            json: false,
+            gzip: false,
+            maxRedirects: 10,
+            followRedirect: true,
+            removeRefererHeader: false,
+            headers: {},
+            method: 'GET',
+            throwOnHttpError: false,
+            abortFunction: () => false,
+        };
+
+        const opts = Object.assign(defaultOptions, options);
+        const method = opts.method.toLowerCase();
+        const request = rqst[method](opts)
+            .on('error', err => reject(err))
+            .on('response', async (res) => {
+                const shouldAbort = opts.abortFunction(res);
+                if (shouldAbort) {
+                    request.abort();
+                }
+
+                // No need to catch invalid content header - it is already caught by request
+                const { type, encoding } = contentType.parse(res);
+
+                // 5XX codes are handled as errors, requests will be retried.
+                const status = res.statusCode;
+                if (status >= 400 && opts.throwOnHttpError) {
+                    let body;
+                    try {
+                        body = await readStreamIntoString(res, encoding);
+                    } catch (err) {
+                        // Error in reading the body.
+                        return reject(err);
+                    }
+                    // Errors are often sent as JSON, so attempt to parse them,
+                    // despite Accept header being set to something different.
+                    if (type === 'application/json') {
+                        const errorResponse = JSON.parse(body);
+                        let { message } = errorResponse;
+                        if (!message) message = util.inspect(errorResponse, { depth: 1, maxArrayLength: 10 });
+                        return reject(new Error(`${status} - ${message}`));
+                    }
+                    // It's not a JSON so it's probably some text. Get the first 100 chars of it.
+                    return reject(new Error(`requestBetter: ${status} - Internal Server Error: ${body.substr(0, 100)}`));
```

Comment:
`utils.requestBetter` ?  Just `requestBetter:` feels like something is missing at the beginning